### ISSUE TITLE
Remote IP addresses aren't being logged correctly

### ISSUE THREAD

**Issue by morungos** [pughlab/tracker#35]
When a user is connected, the application log looks like: 

```
INFO    [org.restlet.SpringComponent.LogService] 2015-08-17	14:11:05	127.0.0.1	ssamadia@ads.uhnresearch.ca	127.0.0.1	80	GET	/api/studies/BRAF/views	-	200	-	-	13	http://clintracker.uhnresearch.ca	Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/44.0.2403.155 Safari/537.36	http://clintracker.uhnresearch.ca/studies/BRAF
```

Note the `127.0.0.1`'s. This isn't a showstopper for security logging, as this is a second-line log, and all accesses will also be logged by nginx. But we should fix it.

**Comment by morungos** [comment]
Question put to the forum, i.e., http://stackoverflow.com/questions/32057807/how-can-i-change-the-default-log-template-in-a-restlet-application

**Comment by morungos** [comment]
Actually, this is probably misdirection, as the actual issue is probably that Jetty doesn't handle this by default, and needs to coerced into using the `X-Forwarded-For` header by a small amount of configuration:  http://www.eclipse.org/jetty/documentation/current/configuring-connectors.html#d0e4404

The HTTP connector should look a bit like this:

    <New id="httpConfig" class="org.eclipse.jetty.server.HttpConfiguration">
      <Set name="outputBufferSize">32768</Set>
      <Set name="requestHeaderSize">8192</Set>
      <Set name="responseHeaderSize">8192</Set>
 
      <Call name="addCustomizer">
        <Arg><New class="org.eclipse.jetty.server.ForwardedRequestCustomizer"/></Arg>
      </Call>
    </New>

This does have an effect, but not a complete one. A log entry now looks like:

```
INFO    [org.restlet.SpringComponent.LogService] 2015-08-18	10:03:47	205.210.128.135	-	127.0.0.1	443	GET	/index.html	-	200	-	-	23	https://clintracker.uhnres.utoronto.ca	Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/44.0.2403.130 Safari/537.36	-
```

So we're getting the IP address of the nginx server, not the full `X-Forwarded-For` chain.

**Comment by morungos** [comment]
This actually suggests maybe the local UHN proxy isn't playing consistent with `X-Forwarded-For`.

**Comment by morungos** [comment]
Ok, that's all I can do. There are issues that remain, but they're network-level issues for the port-forwarding in the deployment environment, and therefore outside the application problems.

**Comment by morungos** [comment]
Here's the analysis results on port forwarding:

If I connect to: http://205.210.128.25/ (no matter if go through the UHN proxy) I get the right remote address. If I connect to http://205.210.128.135 (through the proxy) I get the right address, but if I connect directly via curl, I get Samwise as the remote address. If I connect to http://clintracker.uhnres.utoronto.ca (which doesn’t go through the proxy, but resolves to http://205.210.128.135) then I get Samwise as the remote address. 

In other words, if the connection goes through a proxy, it’s correct. And if it goes through http://205.210.128.25, it’s correct no matter what. The problems are connections to http://205.210.128.135 when they don’t go through a UHN proxy, which they don’t for me.
